components/eam/src/physics/cam/cldera_dynamic_tracers.F90

```diff
@@ -0,0 +1,371 @@
+
+    real(r8) :: pv_out(ncol,pver)
+    if (.not. cldera_dynamic_tracers_flag) then
+      return
+    end if
+    lchnk = state%lchnk
+    do i=1,ncol
+      do k=1,pver
+        pv_out(i,k) = state%q(i,k,ixpv)
+      end do
+    end do
+    call outfld(pv_out_name, pv_out(:,:)-cldera_dynamic_tracers_pv_offset, ncol, lchnk)           
+
+  end subroutine cldera_dynamic_tracers_timestep_tend
+
+! ===========================================================================
+
+  function cldera_dynamic_tracers_is_pv(name)
+    !----------------------------------------------------------------------- 
+    ! 
+    ! Purpose: return true if specified constituent is treated as PV
+    ! 
+    !-----------------------------------------------------------------------
+
+    character(len=*), intent(in) :: name    ! constituent name
+    logical :: cldera_dynamic_tracers_is_pv ! return value
+    cldera_dynamic_tracers_is_pv = (cldera_dynamic_tracers_flag) .and. (name == c_names(1))
+
+
+  end function cldera_dynamic_tracers_is_pv
+
+! ==========================================================================
+ 
+  function cldera_dynamic_tracers_is_pt(name)
+    !----------------------------------------------------------------------- 
+    ! 
+    ! Purpose: return true if specified constituent is treated as PT
+    ! 
+    !-----------------------------------------------------------------------
+
+    character(len=*), intent(in) :: name    ! constituent name
+    logical :: cldera_dynamic_tracers_is_pt ! return value
+    cldera_dynamic_tracers_is_pt = (cldera_dynamic_tracers_flag) .and. (name == c_names(2))
+
+
+  end function cldera_dynamic_tracers_is_pt
+
+! ==========================================================================
+
+ function cldera_dynamic_tracers_is_enabled()
+    !----------------------------------------------------------------------- 
+    ! 
+    ! Purpose: return true if package is activated
+    ! 
+    !-----------------------------------------------------------------------
+
+    logical :: cldera_dynamic_tracers_is_enabled        ! return value
+    cldera_dynamic_tracers_is_enabled = cldera_dynamic_tracers_flag
+
+
+  end function cldera_dynamic_tracers_is_enabled 
+
+!===============================================================================
+  
+  subroutine cldera_dynamic_tracers_init
+
+    !----------------------------------------------------------------------- 
+    ! 
+    ! Purpose: initialize age of air constituents
+    !          (declare history variables)
+    !-----------------------------------------------------------------------
+
+    use cam_history,    only: addfld, add_default, horiz_only
+
+    integer :: m, mm
+    !-----------------------------------------------------------------------
+
+    if (.not. cldera_dynamic_tracers_flag) return
+
+    ! decalre PV, PT tracers as history variables
+    call addfld (pv_out_name, (/ 'lev' /), 'I', 'm2 K/kg/s', cnst_longname(ixpv))
```

Comment:
Why are we handling the names of the PV and PT outputs differently? I might favor just outputting `cnst_name(ixpv)` here for consistency and to avoid additional redirection in naming.

Reply:
I agree with this as well, and didn't understand why this is being done. It seemed to be potentially related to the choice of using `PV_TRCR_` rather than `PV_TRCR`. I suspect it might be a leftover from debugging. @OkayHughes can you please comment?

Reply:
See above comment. This was an artifact of before I understood you can disable "cam_outfld" as an optional argument when the constituent is registered.

Reply:
I opted to leave this as it is; see comments in earlier resolved conversations